kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
+
+      -- user wanted a list of all active targets for this upstream
+      local target_history, err = dao_factory.targets:find_all({
+        upstream_id = self.params.upstream_id,
+      })
+
+      if not target_history then
+        return app_helpers.yield_error(err)
+      end
+
+      -- sort and walk based on target and creation time
+      for _, target in ipairs(target_history) do
+        target.order = target.target..":"..target.created_at
+      end
+      table.sort(target_history, function(a, b) return a.order > b.order end)
```

Comment:
If timestamps are identical, my understanding is that, since the sort keys are identical, the sort order will not change. In this case, the order of the table elements remains unchanged. Consider the following:

```
> t = { { a = 1, foo = "baz" }, { a = 1, foo = "bar" } }
> table.sort(t, function(a, b) return a.a < b.a end)
> print(t[1].foo)
baz
> print(t[2].foo)
bar
```

When we compare on equal values, the order is unchanged.

If multiple target entities contained the _same_ UUID, I agree that's a valid sorting element. But because (from my understanding, please correct me if I'm wrong), each POST to `/upstreams/:name/targets` results in a _new_ entry with a _unique_ UUID, sorting on these UUIDs is useless, and the above applies. If I'm misunderstanding please let me know :)